app/lib/features/chat/providers/chat_providers.dart

```diff
@@ -204,3 +217,50 @@ final chatTypingEventProvider = StreamProvider.autoDispose
         .toList();
   }
 });
+
+// unread notifications, unread mentions, unread messages
+typedef UnreadCounters = (int, int, int);
+
+final unreadCountersProvider = FutureProvider.autoDispose
+    .family<UnreadCounters, String>((ref, roomId) async {
+  final convo = await ref.watch(
+    convoProvider(await ref.watch(chatProvider(roomId).future)).future,
+  );
+  if (convo == null) {
+    return (0, 0, 0);
+  }
+  final ret = (
+    convo.numUnreadNotificationCount(),
+    convo.numUnreadMentions(),
+    convo.numUnreadMessages()
+  );
+  return ret;
+});
+
+final hasUnreadChatsProvider = FutureProvider.autoDispose((ref) async {
+  if (!ref.watch(isActiveProvider(LabsFeature.chatUnread))) {
+    // feature not active
+
+    return UrgencyBadge.none;
+  }
+  final chats = ref.watch(chatsProvider);
+  if (chats.isEmpty) {
+    return UrgencyBadge.none;
+  }
+  UrgencyBadge currentBadge = UrgencyBadge.none;
+
+  for (final chat in chats) {
```

Comment:
Yes, this would seem not a good optimal approach, I am facing similar case in fetching space details for multiple parents. I initially used ```.map()``` approach but somehow the provider is not functioning correctly and only returns one value .i.e. I sense it's because of ```FutureProviders``` bug in terms of dealing with future ```ref.watch()```.  Suggestion would be to use map along with awaiting all futures concurrently and returning first non-null data:
```suggestion
  final badges = await Future.wait(
  chats.map((chat) async {
    final unreadCounter = ref.watch(unreadCountersProvider(chat.getRoomIdStr()).future);
    if (unreadCounter.$1 > 0) {
      return UrgencyBadge.important;
    } else if (unreadCounter.$2 > 0) {
      return UrgencyBadge.unread;
    } else {
      return null;
    }
  }),
);

return badges.firstWhere((badge) => badge!= null, orElse: () => null);
```

Reply:
not sure I agree ... this would poll _all_ providers regardless while mine might be _slower_ but would stop on the first match, which considering that they are cached shouldn't be that much of a difference. I think a better solution is to put all that SDK side, but I leave that for a future implementation I think...